layers.go

```diff
@@ -2468,6 +2475,12 @@ func (r *layerStore) applyDiffWithOptions(to string, layerOptions *LayerOptions,
 	layer.UncompressedDigest = uncompressedDigest
 	layer.UncompressedSize = uncompressedCounter.Count
 	layer.CompressionType = compression
+
+	if layerOptions != nil && layerOptions.LayerFilename != nil {
+		layer.CompressedSize = size
+		layer.UncompressedSize = size
```

Comment:
All of this _just_ doesn’t work; the data was already interpreted as a tar file, and uncompressed. This is at best pretending that didn’t happen.